src/widgets/WidgetMainSmall.vue

```diff
@@ -177,6 +178,11 @@ Indicative monthly payment is a minimum monthly repayment (MMP) of the greater o
 ^ Indicative repayments (Transaction amount divided by Interest Free Period) are an estimate only, which excl $99 Annual Fee, and assumes no additional purchases, cash advances or other fees or charges. Interest Free Period available when indicative monthly repayments are made by each statement period due date, resulting in full repayment of purchase amount within the Interest Free Period. `,
     }
   },
+  beforeCreate() {
+    fetchData('widget', {
+      method: 'POST',
+    })
+  },
```

Comment:
Thanks @arjun-at-roam for walking me through this. If I understand correctly, the requirements is that the project uses Vue 3, but only backwards-compatible features. Here are a few things I'd recommend:

* I'd probably avoid fetching data in the `beforeCreate` hook. This occurs before data observation (I believe this means there's no `this` yet), so if you receive the data too quickly, there's a chance you might not be able to store it in the instance, and then the data will be lost. I believe `created` might be a better option for this. At this point, the instance is up with data, computed, etc, but the mounting phase has not started yet. More info here:

https://v3.vuejs.org/api/options-lifecycle-hooks.html#beforecreate

Just in case, `created` is a lifecycle in Vue 2 as well, so compatibility should not be a problem.

* `vm.$nextTick` is really useful if you're calling something in the `mount` lifecycle, and if you'd like to make sure it happens once the entire view has been rendered (to make sure child components are already up and all). If you're fetching on `create`, you might not need to worry about this .... but if this helps, even better.

https://v3.vuejs.org/api/options-lifecycle-hooks.html#mounted

* I believe all lifecycle hooks in Vue 3 are supported in Vue 2 as well. There should be nothing to worry about unless you're using the Composition API, which does make some of these hooks redundant. I love this diagram from the docs:

https://vuejs.org/v2/guide/instance.html

Hope this helps!

Reply:
Thanks legend @jpyepez-roam !